apps/browser/components/Sidebar.tsx

```diff
@@ -99,6 +104,12 @@ const Sidebar = ({
     }
   };
 
+  useEffect(() => {
+    if (activeChat !== chatId) {
+      setActiveChat(chatId);
+    }
+  }, [chatId, activeChat])
```

Comment:
Isn't chatId already the active chat?
Or is this some trick to get around a react issue?

Reply:
activeChat needs to be update after the global chatId is set, which ensures the chat has been loaded. Yes it creates undesired behavior if we don't do this, the sidebar updates its UI before the app state is ready for it.